src/test/java/com/aws/greengrass/lifecyclemanager/GreengrassServiceTest.java

```diff
@@ -116,4 +133,34 @@ void GIVEN_service_WHEN_dependencies_change_THEN_service_dependencies_updates()
         assertNull(aService.dependencies.get(dService));
         assertNotNull(aService.dependencies.get(eService));
     }
+
+    @Test
+    void GIVEN_service_WHEN_service_is_closed_THEN_service_dependency_watcher_is_removed() throws Exception {
+        // GIVEN service
+        Path configPath = Paths.get(this.getClass().getResource("services.yaml").toURI());
+        Configuration serviceConfiguration = new Configuration(context);
+        serviceConfiguration.read(configPath);
+        Topics root = serviceConfiguration.getRoot();
+        Topics serviceTopics = spy(root.findTopics(SERVICES_NAMESPACE_TOPIC, "B"));
+        Topic externalDependencyTopic = mock(Topic.class);
+
+        doReturn(context).when(serviceTopics).getContext();
+        doReturn(externalDependencyTopic).when(serviceTopics).createLeafChild(eq(SERVICE_DEPENDENCIES_NAMESPACE_TOPIC));
+        doReturn(externalDependencyTopic).when(externalDependencyTopic).dflt(any());
+        doReturn(Collections.emptyList()).when(externalDependencyTopic).getOnce();
+
+        GreengrassService service = spy(new GreengrassService(serviceTopics));
+
+        //WHEN
+        AtomicBoolean watcherRemoved = new AtomicBoolean(false);
+        doAnswer((Answer<Void>) invocationOnMock -> {
+            watcherRemoved.set(true);
+            return null;
+        }).when(externalDependencyTopic).remove(any());
+
+        service.close().get(30, TimeUnit.SECONDS);
+
+        //THEN
+        assertTrue(watcherRemoved.get());
```

Comment:
Does `service.close()` ever get undone expecting the service lifecycle for the same service instance to progress as usual? Or does it always only happen during nucleus shutdown or service removal?

Reply:
The reason I'm asking is to see whether something needs to be tested for that